authentication.md

```diff
@@ -0,0 +1,66 @@
+# Authentication
+
+The underlying SDKs provide Authenticators, which are used for authenticating
+a user in various contexts.  A full list can be found here:
+
+https://github.com/stormpath/stormpath-sdk-spec/blob/master/specifications/authenticators.md
+
+Those authenticators have an interface which is *not* HTTP-specific, the
+interfaces work with the necessary parameters for achieving authentication.
+
+As such, our framework integrations will be responsible for taking an incoming
+HTTP request, gathering the needed authentication parameters (from the body,
+from HTTP headers, etc) and delegating to the necessary authenticator.
+
+This work should be encapsulated into re-usable functions that can be exposed
+in the public API of the framework.
+
+The default set of functions that we want to provide are:
+
+**Account Resolvers**
+
+Account resolvers will check for the presense of authentication information, and return account details from Stormpath if available. This is useful, for instance, for a website that wants to render a menu bar differently for logged in users. 
+
+* `getAccount` - look for all forms of authentication, and load an account if credentials are present. 
+
+**Authentication Filters**
+
+The authentication filters require that valid authentication information is present. 
+
+If the authentication details are not present, these gatekeepers will ALWAYS render an error, rendered in the best content type as according to the Accept headers and produces configuration. 
+
+If `application/json` is preferred: Render `401 Unauthorized` with a blank body.
+
+If `text/html` is preferred: Redirect to `stormpath.web.login.uri`
```

Comment:
And preserve the requested relative URL, by appending it as `<stormpath.web.login.uri>?next=/relative-requtested-url`